lib/spack/spack/build_systems/autotools.py

```diff
@@ -103,7 +103,7 @@ def _do_patch_config_guess(self):
         build date (automake 1.13.4)."""
 
         if not self.patch_config_guess or not self.spec.satisfies(
-                'arch=linux-rhel7-ppc64le'
+                'arch=ppc64le'
```

Comment:
Should this be `target` instead of `arch`?
```
$ spack help --spec
...
    architecture variants:
      target=target                 specific <target> processor
      os=operating_system           specific <operating_system>
      platform=platform             linux, darwin, cray, bgq, etc.
      arch=platform-os-target       shortcut for all three above
```

Reply:
@adamjstewart yes, what I did worked, but your suggestion is much better. I was looking for something more like what you proposed anyway. Thanks!